src/circuit_writer/mod.rs

```diff
@@ -234,18 +240,28 @@ impl<B: Backend> CircuitWriter<B> {
         // get length
         let len = match &typ.kind {
             TyKind::Field => 1,
-            TyKind::Array(typ, len) => {
+            // todo: the size will get from const var
```

Comment:
not sure I understand how actionable that todo is

Reply:
If the ArraySize is a ConstVar, then it can load the const value. This is assume the const variable is in the local module. I think the `ArraySize::ConstVar` would need to be a something like `ExprKind::Variable` that can point to other module's constants if we can support that scenario.

Reply:
(you should write that in the TODO ;))